pipeline_steps/common.groovy

```diff
@@ -27,10 +29,69 @@ def install_ansible(){
     pip install -c ${env.WORKSPACE}/rpc-gating/constraints.txt -U ansible pyrax
 
     mkdir -p rpc-gating/playbooks/roles
-    ansible-galaxy install -r ${env.WORKSPACE}/rpc-gating/role_requirements.yml -p ${env.WORKSPACE}/rpc-gating/playbooks/roles
+    ansible-galaxy install -r rpc-gating/role_requirements.yml -p rpc-gating/playbooks/roles
+
+    # The ansible version used on the Jenkins slave is always new enough for
+    # ara. So ARA will be used to collect data about playbooks run on the
+    # jenkins slave even if the ansible used by osa/rpco is too old. ARA is
+    # installed into the osa ansible runtime venv in branches after mitaka by
+    # rpco/scripts/deploy.sh
+    pip install -e ${env.ARA_REPO}@${env.ARA_BRANCH}#egg=ara
+    mkdir -p callbacks
+    cp .venv/src/ara/ara/plugins/callbacks/log_ara.py callbacks
   """
 }
 
+/* ARA records data about ansible task results,
+ * This function calls ara to generate a junit file from
+ * the ara database, and submits it to jenkins.
+ * This allows jenkins to record each ansible task as a
+ * unit test.
+ * Ara generate will dump all ansible task results from it's db into junit.xml,
+ * to prevent double reporting tests, junit.xml is cleared before and after
+ * submission to jenkins and the ara db is cleared afterwards.
+ */
+def submit_ara(){
+  dir(env.WORKSPACE){
+    db_file="ansible.sqlite"
+    if (fileExists(file: db_file)){
+      withEnv(common.get_deploy_script_env()){
+
+        // remove previously generated junit files and generate a new one.
+        sh """#!/bin/bash -x
+          . .venv/bin/activate
+          rm -f junit.xml ||:
+          ara generate junit junit.xml ||:
```

Comment:
Its not reported explicitly, however the lack of test results displayed in jenkins will alert users to the problem. Also its important these instrumentation functions don't cause build failures as they are not part of the product under test and just there to aid in debugging of legitimate failures. 

The aim of this PR is to reduce the time taken to determine what caused a build to fail, I don't want to introduce more failure cases. If this mechanism fails, and the job itself also fails than debugging can be done as per usual by reading the log.